### ISSUE TITLE
ng-simple-upload.js:42 Uncaught ReferenceError: $ is not defined

### ISSUE THREAD

**Issue by chevalierc** [shammelburg/ng-simple-upload#2]
ng-simple-upload.js:42 Uncaught ReferenceError: $ is not defined

**Comment by shammelburg** [comment]
Have you got jQuery reference in your project? The `README.md` does not contain that it is a dependency, sorry.

**Comment by chevalierc** [comment]
Nah. That what I thought the symbol meant, but didn't think an angular script would be using jquery. They don't always work well together.

**Comment by shammelburg** [comment]
It uses `$.ajax` to post the file to the server. Maybe you can give me more details about which versions you are using? An example of how you are referencing your scripts?

**Comment by chevalierc** [comment]
You can remove the jquery dependency by replacing

```
return $.ajax({
    type: 'POST',
    url: scope.webApiUrl,
    data: fd,
    async: true,
    cache: false,
    contentType: false,
    processData: false
})
```

with

```
return $http.post( scope.webApiUrl, fd, {
            withCredentials: true,
            headers: {
                'Content-Type': undefined
            },
            transformRequest: angular.identity
        }
```

and including $http in the directive

I've been told you shouldn't mix jquery with angular. That why the angular bootstrap exists, so users to use the jquery bootstrap

**Comment by shammelburg** [comment]
Thank you, 

I will test this next time I use it.

As for mixing the two, both are compatible and work fine together. It's more of a personal preference whether you want to use one or both.

Happy coding
